client/app/components/show-project/actions-list/ulurp-number-link.js

```diff
@@ -0,0 +1,28 @@
+import Component from '@ember/component';
+import ENV from 'labs-zap-search/config/environment';
+
+/**
+ * @param {Action Object} action
+ */
+export default class UlurpNumberLinkComponent extends Component {
```

Comment:
I see, you pulled the method from a helper. Why not just make a new helper? Why a component?

Reply:
So now when peple see "build-url" they have to know that ulurp number link is an exception and use a component version of the same logic instead? I don't understand this decision.

Reply:
You can also just export the function from the helper and import it here?

Reply:
> > This PR is bulky because it required componentizing the frontend to introduce some getters, avoiding complex template logic.
> 
> With respect, I find this change actually adds complex template logic.
> 
> Do we know for sure that these urls can't just be computeds on the action model? What can go on the model?

@allthesignals  I like the suggestion to try placing the URL-building (i.e. the serverRelativeUrl computed) on the Action model. I pushed a new commit to do that. It also uses the build-url helper again. I also removed the ulurpLinkNumber component.

Take a look and lmk what you think!

Reply:
I origina

> So now when peple see "build-url" they have to know that ulurp number link is an exception and use a component version of the same logic instead? I don't understand this decision.

I originally broke out the `cpcReport` case from build URL because it was no longer correct. The buildUrl cpcReport worked off of absolute URL, but that was the cause of the bug (links were linking directly to sharepoint). I think my initial reaction was that buildUrl is a legacy component I wanted to stay away from.

Also buildUrl has a long list of "if" conditions, which I feel like is what we want to eventually move away from.